gax/test/test-application/src/index.ts

```diff
@@ -416,11 +417,52 @@ async function testEchoErrorWithRetries(client: SequenceServiceClient) {
   try {
     await client.attemptSequence(attemptRequest, settings);
   } catch (err) {
-    assert.strictEqual(JSON.stringify((err as GoogleError).code), '4');
-    assert.match(
-      JSON.stringify((err as GoogleError).message),
-      /Exceeded maximum number of retries retrying error Error: 14 UNAVAILABLE: 14 before any response was received/,
-    );
+    assert.strictEqual((err as GoogleError).message, 'Exceeded maximum number of retries retrying error Error: 14 UNAVAILABLE: 14 before any response was received : Previous errors : [{message: 14 UNAVAILABLE: 14, code: 14, details: , note: },{message: 14 UNAVAILABLE: 14, code: 14, details: , note: }]');
+    assert.strictEqual((err as GoogleError).code, 4);
```

Comment:
The old assert will pass, but really we should test for a more exact match here.